huxley/api/serializers/delegate.py

```diff
@@ -25,6 +31,32 @@ class Meta:
                   'session_three',
                   'session_four', )
 
+    def update(self, instance, validated_data):
+        if ('assignment' in validated_data and
+                validated_data['assignment'] != None and
+                not instance.assignment):
+
+            names = instance.name.split(' ')
+            username = names[0] + '_' + str(instance.id)
+            password = BaseUserManager().make_random_password(10)
+            user = User.objects.create(delegate=instance, username=username,\
+                                       user_type=User.TYPE_DELEGATE,\
+                                       last_login=datetime.now(),
+                                       first_name=names[0], last_name=names[-1],
+                                       email=instance.email)
+            user.set_password(password)
+            user.save()
```

Comment:
We're hitting the DB twice here: once when we call `create`, and once when we call`save`. We should only do it once (right here). Instead of using the manager's `create` method, you should be able to just construct a User object. If there's a particular reason we're saving twice, feel free to ignore.

Reply:
Good catch; I'll fix this.